components/gitpod-cli/cmd/git-track-command.go

```diff
@@ -0,0 +1,84 @@
+// Copyright (c) 2021 Gitpod GmbH. All rights reserved.
+// Licensed under the GNU Affero General Public License (AGPL).
+// See License-AGPL.txt in the project root for license information.
+
+package cmd
+
+import (
+	"context"
+	"io"
+	"os"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+
+	gitpod "github.com/gitpod-io/gitpod/gitpod-cli/pkg/gitpod"
+	serverapi "github.com/gitpod-io/gitpod/gitpod-protocol"
+)
+
+var gitTrackCommandOpts struct {
+	GitCommand string
+}
+
+var gitTrackCommand = &cobra.Command{
+	Use:    "git-track-command",
+	Short:  "Gitpod's Git command tracker",
+	Long:   "Sending anonymous statistics about the executed git commands inside",
+	Args:   cobra.ExactArgs(0),
+	Hidden: true,
+	Run: func(cmd *cobra.Command, args []string) {
+		log.SetOutput(io.Discard)
+		f, err := os.OpenFile(os.TempDir()+"/gitpod-git-credential-helper.log", os.O_WRONLY|os.O_CREATE|os.O_APPEND, 0644)
+		if err == nil {
+			defer f.Close()
+			log.SetOutput(f)
+		}
+
+		log.Infof("gp git-track-command")
+
+		ctx, cancel := context.WithTimeout(context.Background(), 1*time.Minute)
+		defer cancel()
+		wsInfo, err := gitpod.GetWSInfo(ctx)
+		if err != nil {
+			fail(err.Error())
+		}
+
+		client, err := gitpod.ConnectToServer(ctx, wsInfo, []string{"function:trackEvent"})
```

Comment:
It would be great to provide `User-Agent` and `X-Client-Version` here as additional headers, similarly to https://github.com/gitpod-io/gitpod/blob/fb99a9f5eb9c8c0ce9955f8eefa92445352644b8/components/local-app/main.go#L264-L266

Reply:
What should be the UA? `gitpod/workspace`?

Reply:
`gitpod/cli`

Reply:
Does this mean I should rewrite this segment to use the `gitpod` client from the `gitpod-protocol` instead of `gitpod-cli`? It seems that the one we are using here right now doesn't accept any more params.

Reply:
It should be the same client, maybe function should be extended to allow passing headers.